yjit/src/backend/arm64/mod.rs

```diff
@@ -821,7 +827,11 @@ impl Assembler
                     // the Arm64 assembler works, the register that is going to
                     // be stored is first and the address is second. However in
                     // our IR we have the address first and the register second.
-                    stur(cb, src.into(), dest.into());
+                    match dest.rm_num_bits() {
+                        64 => stur(cb, src.into(), dest.into()),
+                        32 => sturh(cb, src.into(), dest.into()),
+                        _ => panic!("unexpected dest num_bits: {}, src: {:#?}, dest: {:#?}", dest.rm_num_bits(), src, dest),
```

Comment:
We could reuse the result of `rm_num_bits()`

```suggestion
                        num_bits => panic!("unexpected dest num_bits: {} (src: {:#?}, dest: {:#?})", num_bits, src, dest),
```